design/KEPS/multicluster/proposal.md

```diff
@@ -0,0 +1,160 @@
+
+1. Creating a service account in the target cluster.
+2. Granting proper permissions for this service account.
+3. Creating a secret with the credentials for this service account in the cluster/namespace where Kiali is deployed.
+
+This process will need to be repeated for every new cluster added. Adding or removing clusters will also require restarting the Kiali pod in order for the cache watches to pick up the new kube config. Adding/removing kube clusters without having to restart Kiali can be an improvement made later.
+
+## Operator
+
+The Kiali operator will still be used for configuring Kiali itself however, it's assumed that the operator won't have access to other kube clusters other than where Kiali is deployed and therefore configuring multicluster access for Kiali will be a manual operation. The operator will remain unchanged in a multi-cluster deployment.
+
+## Testing
+
+All automated e2e testing, API and cypress, is currently performed against a single cluster in CI. There will be a large number of Kiali features related to multi-cluster so there needs to be a way to test multi-cluster deployments frequently and in an automated fashion. Any e2e tests that require a multi-cluster environment will need to be run separately from the rest of the suite.
+
+## Other solutions
+
+### Kiali Agent
+
+A process that runs on each kube cluster that is a part of the mesh and exports Kiali’s view of that mesh to a centralized Kiali. This option offers potential performance gains and ease of configuration since Kiali no longer has to communicate with each individual kube cluster. However, it complicates Kiali’s ability to perform authorization with RBAC since the agents would be exporting data to a centralized Kiali and the centralized Kiali would not have access to the individual clusters. The agent itself adds an additional component to Kiali that would need to be maintained and configured across clusters.
+
+### Connecting directly to istiod
+
+The istiods provide istio config and some information about the mesh workloads. However, the endpoints are not always exposed outside of the istiod pod. Kiali today uses port forwarding through the kubernetes API to connect to the istiod pods and this would require access to the kube API anyways. Some other features rely on the kube API like streaming logs and any create/update operations need to use the Kube API ultimately making this an un-viable option to keep most of the existing Kiali features.
+
+# Roadmap
+
+- [ ] Multicluster Kubernetes support for Kiali.
+  - [ ] Cache can be configured for multiple kubernetes clusters.
+  - [ ] Client can be configured for multiple kubernetes clusters.
+- [ ] Business services query across configured clusters for resources.
+- [ ] Investigate automated testing setup: how kiali should test multi-cluster features in CI/CD.
+- [ ] Multi-cluster demos for the different deployment models. These can also be used for testing.
```

Comment:
Maybe add a new point regarding the UI changes? Something like: 
[] Define UI/UX changes to support MC